diff --git a/front_end/timeline/TimelinePanel.js b/front_end/timeline/TimelinePanel.js
index 50348771e..eb0040bf1 100644
--- a/front_end/timeline/TimelinePanel.js
+++ b/front_end/timeline/TimelinePanel.js
@@ -757,41 +757,8 @@ export class TimelinePanel extends UI.Panel.Panel {
       return;
     }
 
-    /**
-     * @param {string} tagName
-     * @param {string} contents
-     */
-    function encloseWithTag(tagName, contents) {
-      const e = createElement(tagName);
-      e.textContent = contents;
-      return e;
-    }
-
-    const learnMoreNode = UI.XLink.XLink.create(
-        'https://developers.google.com/web/tools/chrome-devtools/evaluate-performance/',
-        Common.UIString.UIString('Learn\xa0more'));
-
-    const recordKey =
-        encloseWithTag('b', self.UI.shortcutRegistry.shortcutsForAction('timeline.toggle-recording')[0].title());
-    const reloadKey =
-        encloseWithTag('b', self.UI.shortcutRegistry.shortcutsForAction('timeline.record-reload')[0].title());
-    const navigateNode = encloseWithTag('b', Common.UIString.UIString('WASD'));
-
     this._landingPage = new UI.Widget.VBox();
     this._landingPage.contentElement.classList.add('timeline-landing-page', 'fill');
-    const centered = this._landingPage.contentElement.createChild('div');
-
-    const recordButton = UI.UIUtils.createInlineButton(UI.Toolbar.Toolbar.createActionButton(this._toggleRecordAction));
-    const reloadButton =
-        UI.UIUtils.createInlineButton(UI.Toolbar.Toolbar.createActionButtonForId('timeline.record-reload'));
-
-    centered.createChild('p').appendChild(UI.UIUtils.formatLocalized(
-        'Click the record button %s or hit %s to start a new recording.\nClick the reload button %s or hit %s to record the page load.',
-        [recordButton, recordKey, reloadButton, reloadKey]));
-
-    centered.createChild('p').appendChild(UI.UIUtils.formatLocalized(
-        'After recording, select an area of interest in the overview by dragging.\nThen, zoom and pan the timeline with the mousewheel or %s keys.\n%s',
-        [navigateNode, learnMoreNode]));
 
     this._landingPage.show(this._statusPaneContainer);
   }
