diff --git a/front_end/profiler/module.json b/front_end/profiler/module.json
index 4b27a675..70583e3c 100644
--- a/front_end/profiler/module.json
+++ b/front_end/profiler/module.json
@@ -1,13 +1,5 @@
 {
     "extensions": [
-        {
-            "type": "view",
-            "location": "panel",
-            "id": "heap_profiler",
-            "title": "Memory",
-            "order": 60,
-            "className": "Profiler.HeapProfilerPanel"
-        },
         {
             "type": "view",
             "location": "drawer-view",
@@ -49,7 +41,7 @@
             "toggledIconClass": "largeicon-stop-recording",
             "toggleWithRedColor": true,
             "className": "Profiler.LiveHeapProfileView.ActionDelegate",
-            "category": "Memory",
+            "category": "Performance",
             "options": [
                 {
                     "value": true,
@@ -66,15 +58,13 @@
             "actionId": "live-heap-profile.start-with-reload",
             "iconClass": "largeicon-refresh",
             "className": "Profiler.LiveHeapProfileView.ActionDelegate",
-            "category": "Memory",
+            "category": "Performance",
             "title": "Start recording heap allocations and reload the page"
         },
         {
             "type": "action",
             "actionId": "profiler.heap-toggle-recording",
-            "category": "Memory",
             "iconClass": "largeicon-start-recording",
-            "title": "Start/stop recording",
             "toggleable": true,
             "toggledIconClass": "largeicon-stop-recording",
             "toggleWithRedColor": true,
@@ -96,8 +86,6 @@
         {
             "type": "action",
             "actionId": "profiler.js-toggle-recording",
-            "category": "JavaScript Profiler",
-            "title": "Start/stop recording",
             "iconClass": "largeicon-start-recording",
             "toggleable": true,
             "toggledIconClass": "largeicon-stop-recording",
diff --git a/front_end/timeline/TimelinePanel.js b/front_end/timeline/TimelinePanel.js
index 52b47d1c..7e31e41f 100644
--- a/front_end/timeline/TimelinePanel.js
+++ b/front_end/timeline/TimelinePanel.js
@@ -148,6 +148,9 @@ export class TimelinePanel extends UI.Panel.Panel {
         Extensions.ExtensionServer.Events.TraceProviderAdded, this._appendExtensionsToToolbar, this);
     SDK.SDKModel.TargetManager.instance().addEventListener(
         SDK.SDKModel.Events.SuspendStateChanged, this._onSuspendStateChanged, this);
+
+    this._panelToolbar.element.style = 'display: none';
+    this._panelRightToolbar.element.style = 'display: none';
   }
 
   /**
@@ -749,41 +752,9 @@ export class TimelinePanel extends UI.Panel.Panel {
       return;
     }
 
-    /**
-     * @param {string} tagName
-     * @param {string} contents
-     */
-    function encloseWithTag(tagName, contents) {
-      const e = createElement(tagName);
-      e.textContent = contents;
-      return e;
-    }
-
-    const learnMoreNode = UI.XLink.XLink.create(
-        'https://developers.google.com/web/tools/chrome-devtools/evaluate-performance/',
-        Common.UIString.UIString('Learn\xa0more'));
-
-    const recordKey =
-        encloseWithTag('b', self.UI.shortcutRegistry.shortcutDescriptorsForAction('timeline.toggle-recording')[0].name);
-    const reloadKey =
-        encloseWithTag('b', self.UI.shortcutRegistry.shortcutDescriptorsForAction('timeline.record-reload')[0].name);
-    const navigateNode = encloseWithTag('b', Common.UIString.UIString('WASD'));
-
     this._landingPage = new UI.Widget.VBox();
     this._landingPage.contentElement.classList.add('timeline-landing-page', 'fill');
-    const centered = this._landingPage.contentElement.createChild('div');
-
-    const recordButton = UI.UIUtils.createInlineButton(UI.Toolbar.Toolbar.createActionButton(this._toggleRecordAction));
-    const reloadButton =
-        UI.UIUtils.createInlineButton(UI.Toolbar.Toolbar.createActionButtonForId('timeline.record-reload'));
-
-    centered.createChild('p').appendChild(UI.UIUtils.formatLocalized(
-        'Click the record button %s or hit %s to start a new recording.\nClick the reload button %s or hit %s to record the page load.',
-        [recordButton, recordKey, reloadButton, reloadKey]));
 
-    centered.createChild('p').appendChild(UI.UIUtils.formatLocalized(
-        'After recording, select an area of interest in the overview by dragging.\nThen, zoom and pan the timeline with the mousewheel or %s keys.\n%s',
-        [navigateNode, learnMoreNode]));
 
     this._landingPage.show(this._statusPaneContainer);
   }
